pkg/plugin/work-in-progress/event_handler.go

```diff
@@ -178,3 +146,89 @@ func (gh *GitHubWIPPRHandler) checkTitleAndSetStatus(log log.Logger, pullRequest
 	return statusService.Success(ReadyForReviewMessage, ReadyForReviewDetailsPageName)
 
 }
+
+func (gh *GitHubWIPPRHandler) updateTitleAndResetStatus(log log.Logger, pullRequest *gogh.PullRequest) error {
+	change := gh.getRepositoryChange(pullRequest)
+	statusService := gh.getStatusService(log, change)
+	labels := gh.getPullRequestLabels(log, change, pullRequest)
+	configuration := LoadConfiguration(log, change)
+
+	labelExists := gh.hasWorkInProgressLabel(labels, configuration.Label)
+	if !labelExists {
+		if prefix, ok := gh.IsWorkInProgress(*pullRequest.Title, configuration); ok {
+			*pullRequest.Title = strings.TrimPrefix(*pullRequest.Title, prefix)
+			err := gh.Client.EditPullRequest(change, *pullRequest.Number, pullRequest)
+			if err != nil {
+				log.Warnf("failed to update PR title [%q]. cause: %s", *pullRequest, err)
+			}
+		}
+		return statusService.Success(ReadyForReviewMessage, ReadyForReviewDetailsPageName)
+	}
+	return statusService.Failure(InProgressMessage, InProgressDetailsPageName)
+}
+
+func (gh *GitHubWIPPRHandler) checkLabelAndSetStatus(log log.Logger, pullRequest *gogh.PullRequest) error {
+	change := gh.getRepositoryChange(pullRequest)
+	statusService := gh.getStatusService(log, change)
+	labels := gh.getPullRequestLabels(log, change, pullRequest)
+	configuration := LoadConfiguration(log, change)
+
+	labelExists := gh.hasWorkInProgressLabel(labels, configuration.Label)
+	if labelExists {
+		return statusService.Failure(InProgressMessage, InProgressDetailsPageName)
+	}
+	return statusService.Success(ReadyForReviewMessage, ReadyForReviewDetailsPageName)
+}
+
+func (gh *GitHubWIPPRHandler) getPullRequestLabels(log log.Logger, change scm.RepositoryChange, pullRequest *gogh.PullRequest) []*gogh.Label {
+	labels, err := gh.Client.ListPullRequestLabels(change, *pullRequest.Number)
+	if err != nil {
+		log.Warnf("failed to list labels on PR [%q]. cause: %s", *pullRequest, err)
```

Comment:
log it as an error.
btw, I think that you should call `err.Error()` to get the message